src/test/java/com/google/cloud/spanner/pgadapter/python/PythonTransactionTests.java

```diff
@@ -0,0 +1,1095 @@
+            .getTransaction()
+            .getId()
+            .toStringUtf8();
+
+    Assert.assertEquals(true, transactionIdForSql2.equals(transactionIdForSql3));
+
+    String transactionIdForSql5 =
+        requests.stream()
+            .filter(req -> req.getSql().equals(sql5))
+            .findAny()
+            .get()
+            .getTransaction()
+            .getId()
+            .toStringUtf8();
+    String transactionIdForSql6 =
+        requests.stream()
+            .filter(req -> req.getSql().equals(sql6))
+            .findAny()
+            .get()
+            .getTransaction()
+            .getId()
+            .toStringUtf8();
+
+    Assert.assertEquals(true, transactionIdForSql5.equals(transactionIdForSql6));
+  }
+
+  @Test
+  public void testAutocommitInTransactions() throws IOException, InterruptedException {
+    // tests autocommit settings using connection.autocommit variable
+    List<String> statements = new ArrayList<>();
+    String sql1 = "Select * from some_table";
+    String sql2 = "insert into some_table(col1, col2) values(value1, value2)";
+    String sql3 = "Select * from some_table3";
+    String sql4 = "insert into some_table3(col1, col2) values(value1, value2)";
+
+    // no commit request will be sent after this query because autocommit is false
+    statements.add("query");
+    statements.add(sql1);
+
+    // no commit request will be sent after this update because autocommit is false
+    statements.add("update");
+    statements.add(sql2);
+
+    // no commit request will be sent after this update because autocommit is false
+    statements.add("update");
+    statements.add(sql4);
+
+    // 1 commit request will be sent
+    statements.add("transaction");
+    statements.add("commit");
+
+    statements.add("transaction");
+    statements.add("set autocommit True");
+
+    // no commit request after this query although autocommit is true because it is a query sql
+    statements.add("query");
+    statements.add(sql3);
+
+    // 1 commit request will be sent after this update because autocommit is true
+    statements.add("update");
+    statements.add(sql2);
+
+    // 1 commit request will be sent after this update because autocommit is true
+    statements.add("update");
+    statements.add(sql4);
+
+    mockSpanner.putStatementResult(
+        StatementResult.query(Statement.of(sql1), createResultSet(1, "abcd")));
+    mockSpanner.putStatementResult(StatementResult.update(Statement.of(sql2), 1));
+    mockSpanner.putStatementResult(
+        StatementResult.query(Statement.of(sql3), createResultSet(2, "pqrs")));
+    mockSpanner.putStatementResult(StatementResult.update(Statement.of(sql4), 2));
+
+    String expectedOutput = "(1, 'abcd')\n" + "1\n" + "2\n" + "(2, 'pqrs')\n" + "1\n" + "2\n";
+    String actualOutput = executeTransactions(pgServer.getLocalPort(), statements);
+    for (AbstractMessage request : mockSpanner.getRequests())
+      System.out.println(request.getClass());
```

Comment:
nit: I think this should be removed

Reply:
Thanks for pointing this out. Removed it